Dont allow blank patterns in Dic Entry

feerrenrut opened the pull request:
A blank pattern in a dictionary entry causes issues, see #6412

feerrenrut committed:
Dont allow blank patterns in Dic Entry

A blank pattern in a dictionary entry causes issues see, #6412

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1277,6 +1277,10 @@ def getType(self):
 	def onOk(self,evt):
+		if not self.patternTextCtrl.GetValue():
+			# Translators: This is an error message to let the user know that the pattern field in the dictionary entry is not valid.
+			gui.messageBox(_("A pattern is required."), _("Dictionary Entry Error"), wx.OK|wx.ICON_WARNING, self)
```
Would it be a better user experience to disable the OK button when there is no text in the pattern field?

feerrenrut committed:
move focus to empty pattern field on ok

After showing a warning about the empty pattern field, move the focus to
the field so that the user can update it.
